File: todoapp/src/main/java/com/eltonkola/todoapp2/TodoApp.java
package com.eltonkola.todoapp2;

import com.eltonkola.arkitekt.ArkitektApp;
import com.eltonkola.config.AppRoute;
import com.eltonkola.todoapp2.data.ToDoRepository;

public class TodoApp extends ArkitektApp {

    public static TodoApp app(){
        return (TodoApp) Companion.app();
    }

    private ToDoRepository mToDoRepository;

    public ToDoRepository getToDoRepository(){
        return  mToDoRepository;
    }


    @Override
    public void onCreate() {
        super.onCreate();

        mToDoRepository =  new ToDoRepository(this);
        routeConfig(AppRoute.routes);

    }
}

File: todoapp/src/main/java/com/eltonkola/todoapp2/TodoApp.kt
package com.eltonkola.todoapp2

import com.eltonkola.arkitekt.ArkitektApp
import com.eltonkola.config.AppRoute
import com.eltonkola.todoapp2.data.ToDoRepository
import java.util.*

class TodoApp : ArkitektApp() {

    lateinit var toDoRepository: ToDoRepository
        private set


    override fun onCreate() {
        super.onCreate()

        toDoRepository = ToDoRepository(this)
        routeConfig(AppRoute.routes as HashMap<String, Class<*>>)

    }

    companion object {

        fun getApp(): TodoApp {
            return app() as TodoApp
        }
    }


}
